core/src/main/java/com/datastax/oss/driver/internal/core/CqlIdentifiers.java

```diff
@@ -16,21 +16,36 @@
 package com.datastax.oss.driver.internal.core;
 
 import com.datastax.oss.driver.api.core.CqlIdentifier;
+import com.datastax.oss.driver.shaded.guava.common.base.Preconditions;
 import com.datastax.oss.driver.shaded.guava.common.collect.ImmutableList;
 import com.datastax.oss.driver.shaded.guava.common.collect.ImmutableMap;
+import edu.umd.cs.findbugs.annotations.NonNull;
 import java.util.List;
 import java.util.Map;
+import java.util.function.Function;
 
 public class CqlIdentifiers {
 
-  public static List<CqlIdentifier> wrap(Iterable<String> in) {
+  public static List<CqlIdentifier> wrap(
+      @NonNull Iterable<String> in, @NonNull Function<String, CqlIdentifier> fn) {
+
+    Preconditions.checkNotNull(in, "Input Iterable must not be null");
+    Preconditions.checkNotNull(fn, "CqlIdentifier conversion function must not be null");
     ImmutableList.Builder<CqlIdentifier> out = ImmutableList.builder();
```

Comment:
I think that to be consistent the result of apply `fn.apply(name)` should also be checked using 
`Preconditions.checkNotNull` but maybe it is not needed?

Reply:
Guava's immutable list builder will definitely be unhappy if we try to add a null value here.  But this isn't a problem at the moment, at least as the code is currently written.  CqlIdentifier.fromCql() is annotated with @NonNull; presumably that's why the old impl was safe to use it without such a check.  And since CqlIdentifier.fromInternal() is also marked with @NonNull the same logic should apply for all cases of the new impl.

Obviously if those guarantees go away things become problematic... but at least at the moment it isn't a concern.